openmetadata-service/src/main/java/org/openmetadata/service/search/SearchRepository.java

```diff
@@ -0,0 +1,406 @@
+            "",
+            entity.getEntityReference().getType() + ".id",
+            ENTITY_TO_CHILDREN_MAPPING.get(entity.getEntityReference().getType()).toString());
+    }
+  }
+
+  default void handleSoftDeletedAndRestoredEntity(EntityInterface entity, boolean delete) {
+    if (entity.getEntityReference().getType().equals(Entity.DATABASE_SERVICE)) {
+      this.softDeleteOrRestoreChildrenFromSearch(
+          entity,
+          delete,
+          "service.id",
+          ENTITY_TO_CHILDREN_MAPPING.get(entity.getEntityReference().getType()).toString());
+    } else {
+      this.softDeleteOrRestoreChildrenFromSearch(
+          entity,
+          delete,
+          entity.getEntityReference().getType() + ".id",
+          ENTITY_TO_CHILDREN_MAPPING.get(entity.getEntityReference().getType()).toString());
+    }
+  }
+
+  boolean createIndex(ElasticSearchIndexType elasticSearchIndexType, String lang);
+
+  void updateIndex(ElasticSearchIndexType elasticSearchIndexType, String lang);
+
+  void deleteIndex(ElasticSearchIndexType elasticSearchIndexType);
+
+  Response search(SearchRequest request) throws IOException;
+
+  Response searchBySourceUrl(String sourceUrl) throws IOException;
+
+  Response searchByField(String fieldName, String fieldValue, String index) throws IOException;
+
+  Response aggregate(String index, String fieldName, String value, String query) throws IOException;
+
+  Response suggest(SearchRequest request) throws IOException;
+
+  ElasticSearchConfiguration.SearchType getSearchType();
+
+  default void updateElasticSearch(UpdateRequest updateRequest) throws IOException {
+    throw new CustomExceptionMessage(Response.Status.NOT_IMPLEMENTED, NOT_IMPLEMENTED_METHOD);
+  }
+
+  default void updateElasticSearch(org.elasticsearch.action.update.UpdateRequest updateRequest) throws IOException {
+    throw new CustomExceptionMessage(Response.Status.NOT_IMPLEMENTED, NOT_IMPLEMENTED_METHOD);
+  }
+
+  void updateSearchEntityCreated(EntityInterface entity);
+
+  void updateSearchEntityCreated(EntityTimeSeriesInterface entity);
+
+  void deleteByScript(String index, String scriptTxt, HashMap<String, Object> params);
+
+  void updateSearchEntityDeleted(EntityInterface entity, String script, String field, String alias);
+
+  void softDeleteOrRestoreEntityFromSearch(EntityInterface entity, boolean delete);
+
+  void softDeleteOrRestoreChildrenFromSearch(EntityInterface entity, boolean delete, String field, String alias);
+
+  void updateSearchEntityUpdated(EntityInterface entity, String script, String field);
+
+  void updateSearchChildrenUpdated(
+      EntityInterface entity, String scriptTxt, String field, String value, String alias, Object data);
+
+  void close();
+
+  default BulkResponse bulk(BulkRequest data, RequestOptions options) throws IOException {
+    throw new CustomExceptionMessage(Response.Status.NOT_IMPLEMENTED, NOT_IMPLEMENTED_METHOD);
+  }
+
+  default org.elasticsearch.action.bulk.BulkResponse bulk(
```

Comment:
we are defaulting to elasticsearch classes?

Reply:
we have both declared .. for some reason we did this way .. exactly not recollecting